### ISSUE TITLE
Proposal: Add a standard `config` directory, format, and naming structure

### ISSUE THREAD

**Issue by markbates** [gobuffalo/buffalo#1153]
As Buffalo works to make tooling and plugins easier to write, there is going to become a need for these tools to be able to configure themselves, and to configure them for use with Buffalo.

It is also helpful to app developers to have a standard place to put their configs for their apps as well.

To this end I propose a new `config/` folder in applications. This folder will automatically be bundled with `buffalo build` so tools will still have access to them at right time. app developers can also access them as well using `packr.NewBox("<path to config/>").

I believe all need a common file format and naming system. I propose TOML for the file format and `config/<tool-name>.toml` for the file name. 

**"External" Tools:**
Example `config/vendor/buffalo-heroku.toml`.
Example `config/vendor/buffalo-azure.toml`.

**"Buffalo/Internal" Tools:**
Example `config/buffalo/dev.toml`.
Example `config/buffalo/build.toml`.

**Reserved names in `config/` are**:
`app.toml`
`buffalo.toml`

**Comment by gedw99** [comment]
Like..

So the env approach works in tandem with this ? Or they are for different aspects ?

**Comment by markbates** [comment]
Yes, the `.env` is for configuring environment variables. this folder is meant mostly for tool developers to have a common place to configurations for their tools. think, for example, `buffalo dev` or `buffalo test`, etc... 

It would also, but it being there, would be a place that app developers could leverage to put any configuration files they might want.

**Comment by gedw99** [comment]
Got it. Thanks Mark
